### ISSUE TITLE
Matching POW jobs to recipes

### ISSUE THREAD

**Issue by USGSsakins** [USGS-Astrogeology/PDS-Pipelines#186]
When processing POW jobs, there is no instrument name that matches the recipe JSON files.  To work around this, the URLs for the files being processed by POW can be used to lookup the recipes using the https://github.com/USGS-Astrogeology/PDS-Pipelines/blob/master/pds_pipelines/PDSinfo.json.

1. The URL needs to have the protocol, hostname, and `mission` stripped from the front of the URL and the filename stripped from the end.
2. Next, prepend `/pds_san/PDS_Archives/` to the front of the URL
3. the PDSinfo.json includes a `path` variable in each block, use this to find the matching path to lookup the recipe name

**Comment by AustinSanders** [comment]
@dpmayerUSGS @USGSsakins Stripping the beginning of the filename proved to be pretty easy, but it's not so easy to programmatically strip the filename stuff at the end of the filename.  It's hard to tell which parts of the filename are part of the 'path' and which are filename specific.  An example is as follows:
```
http://pdsimage.wr.usgs.gov/Missions/Galileo/SSI/go_0017/g1/europa/c034987/5126r.img
```
Is identified by
```
 "galileo_ssi_edr": {
    "archiveid": "25",
     "path": "/pds_san/PDS_Archive/Galileo/SSI/"
},

```
In this example, everything after  '/SSI/' is specific to the file -- we can't just strip off the 5126r.img portion.  The file-specific information is naturally highly mission / instrument specific, so I'm not sure there's a way to cleanly strip it off.

To solve this problem, I used:
```
if filename.startswith(pds_info_dict[key]['path']):
   # Where 'key' is the PDSinfo.json clean name
   return(key)
```

The problem with this (and the original proposed solution of stripping filenames) is that paths do not uniquely identify clean names, e.g.
```
    "mro_hirise": {
        "archiveid": "71",
        "path": "/pds_san/PDS_Archive/Mars_Reconnaissance_Orbiter/HiRISE/",
	"bandbinQuery": "Name"
    },
    "mro_hirise_edr": {
        "archiveid": "124",
        "path": "/pds_san/PDS_Archive/Mars_Reconnaissance_Orbiter/HiRISE/",
	"bandbinQuery": "Name",
	"upc_reqs": [".IMG"]
    },
    "mro_hirise_extras": {
        "archiveid": "126",
        "path": "/pds_san/PDS_Archive/Mars_Reconnaissance_Orbiter/HiRISE/"
    },
    "mro_hirise_rdr": {
        "archiveid": "125",
        "path": "/pds_san/PDS_Archive/Mars_Reconnaissance_Orbiter/HiRISE/"
    },

```

As implemented, this will nondeterministically choose a clean name that matches the path specification and use the associated recipe.  If all matching clean names have existing recipes, and all those recipes match, then we should be ok, but that's a pretty broad assumption.

-------------- 

TLDR:

This functionality is implemented in my branch, but there are some assumptions that need to be checked.
